python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -34,6 +36,47 @@ def __init__(self, config: Mapping[str, str], log):  # pylint: disable=too-many-
         # snowflake.connector.connect() because they will override the default values set within the
         # connector; remove them from the conn_args dict.
         self.connector = config.get("connector", None)
+        self.sqlalchemy_engine_args = {}
+
+        # there are three different ways to authenticate with snowflake, we need to ensure that only
+        # one method is provided
+        auths_set = 0
```

Comment:
Is there a better or more pythonic way to do a check that one and only one of three conditions is met?